app/scripts/controllers/metametrics.ts

```diff
@@ -1085,8 +1234,9 @@ export default class MetaMetricsController {
     // event that relies on this promise being fulfilled before performing UI
     // updates, or otherwise delaying user interaction, supply the
     // 'flushImmediately' flag to the trackEvent method.
-    return new Promise((resolve, reject) => {
-      const callback = (err) => {
+    return new Promise<void>((resolve, reject) => {
+      // eslint-disable-next-line @typescript-eslint/no-explicit-any
```

Comment:
Can we use `err: unknown` instead of `any`?

I can't give a suggestion here because it covers lines you haven't changed, but what about this:

``` typescript
      const callback = (err: unknown) => {
        if (err) {
          const message = isErrorWithMessage(err) ? err.message : '';
          const stack = isErrorWithStack(err) ? err.stack : undefined;
          // The error that segment gives us has some manipulation done to it
          // that seemingly breaks with lockdown enabled. Creating a new error
          // here prevents the system from freezing when the network request to
          // segment fails for any reason.
          const safeError = new Error(message);
          if (stack) {
            safeError.stack = stack;
          }
          return reject(safeError);
        }
        return resolve();
      };
```